src/azul/deployment.py

```diff
@@ -480,7 +480,11 @@ def _log_client_response(self,
                              **kwargs: Any
                              ) -> Optional[AWSResponse]:
         event_name = self._shorten_event_name(event_name, self._response_event_name)
-        response = kwargs['response_dict']
+        response, exception = kwargs.get('response_dict'), kwargs.get('exception')
+        if response is None and exception is not None:
```

Comment:
If I understood the botocore code correctly, these values are mutually exclusive…

>https://github.com/boto/botocore/blob/febc9ea08510c8b90c67b456f85a9faa64a69c69/botocore/endpoint.py#L253
>
```python
    def _get_response(self, request, operation_model, context):
        # This will return a tuple of (success_response, exception)
        # and success_response is itself a tuple of
        # (http_response, parsed_dict).
        # If an exception occurs then the success_response is None.
        # If no exception occurs then exception is None.
        success_response, exception = self._do_get_response(
            request, operation_model, context
        )
        kwargs_to_emit = {
            'response_dict': None,
            'parsed_response': None,
            'context': context,
            'exception': exception,
        }
        if success_response is not None:
            http_response, parsed_response = success_response
            kwargs_to_emit['parsed_response'] = parsed_response
            kwargs_to_emit['response_dict'] = convert_to_response_dict(
                http_response, operation_model
            )
        service_id = operation_model.service_model.service_id.hyphenize()
        self._event_emitter.emit(
            f"response-received.{service_id}.{operation_model.name}",
            **kwargs_to_emit,
        )
        return success_response, exception

```

Reply:
Discussed in PL.